.github/workflows/ci.yml

```
@@ -2,11 +2,13 @@
 
 name: CI
 
-on: [push, pull_request]
+on:
+  pull_request:
+    types: [opened, reopened, ready_for_review]
+  push:
```

this will still trigger multiple because you are not specifing a branch.  From what I recall if you do 

```suggestion
  push:
    branches:
      - 'main'
```

that will still trigger. the ci will still trigger on each push to a PR because it is opened and ready for review